pkg/pubnet_ipam/common.go

```diff
@@ -0,0 +1,187 @@
+package ovs_subnet
+
+import (
+	"fmt"
+	log "github.com/golang/glog"
+	"net"
+	"sync"
+	"strings"
+	"path"
+
+	"github.com/openshift/openshift-sdn/pkg/netutils"
+	"github.com/coreos/go-etcd/etcd"
+
+	netutils_server "github.com/openshift/openshift-sdn/pkg/netutils/server"
+)
+
+type PubnetIpam struct {
+	pnIpam          *netutils.IPAllocator
+	gateway		string
+	mux             sync.Mutex
+	etcdCli		*etcd.Client
+	etcdPath	string
+}
+
+func NewPubnetIpamServer(publicNetwork string,
+                         etcdPeers []string,
+                         etcdPath string,
+                         etcdCertfile string,
+                         etcdKeyfile string,
+                         etcdCafile string) (*PubnetIpam, error) {
```

Comment:
The etcd portion needs to be split out into an api and a registry. The api will define the methods needed and the registry will actually implement using etcd.

Just the same way as ovssubnet uses the pkg/api/types.go:<SubnetRegistry>
Create a pkg/api/types.go:<IpvlanRegistry>